src/user-request-error.ts

```diff
@@ -1,25 +1,30 @@
+import ModbusAbstractRequest from './abstract-request'
 import ModbusAbstractResponse from './abstract-response'
 
 export type UserRequestErrorCodes = 'OutOfSync' | 'Protocol' | 'Timeout' | 'ManuallyCleared' | 'ModbusException' | 'Offline' | 'crcMismatch'
 
-export interface IUserRequestError<Res extends ModbusAbstractResponse> {
+export interface IUserRequestError<Res extends ModbusAbstractResponse, Req extends ModbusAbstractRequest> {
   err: UserRequestErrorCodes
   message: string
-  response?: Res
+  response?: Res,
+  request?: Req
 }
 
-export class UserRequestError<Res extends ModbusAbstractResponse> implements IUserRequestError<Res> {
+export class UserRequestError<Res extends ModbusAbstractResponse, Req extends ModbusAbstractRequest>
+    implements IUserRequestError<Res, Req> {
   public err: UserRequestErrorCodes
   public message: string
+  public request?: Req
   public response?: Res
-  constructor ({ err, message, response }: IUserRequestError<Res>) {
+  constructor ({ err, message, request, response }: IUserRequestError<Res, Req>) {
```

Comment:
I've changed the order as requested, but I believe since the parameters are being passed as an object, the order that is being specified here doesn't matter for the enduser.

Also writing the following produces an error:

```js
{ err, message, response, request? }
```

Reply:
Ahh, I am very sorry, I did not notice that they where passed in as a object. Too much C++ lately.